core/server/data/migration/populate.js

```diff
@@ -30,20 +32,31 @@ populate = function populate(options) {
             context: {
                 internal: true
             }
-        },
-        tableSequence = Promise.mapSeries(schemaTables, function createTable(table) {
-            logger.info('Creating table: ' + table);
-            return commands.createTable(table);
-        });
+        };
 
     logger.info('Creating tables...');
 
-    if (tablesOnly) {
-        return tableSequence;
-    }
+    return new Promise(function populateDatabase(resolve, reject) {
+        db.knex.transaction(function populateDatabaseInTransaction(transaction) {
+            return Promise.mapSeries(schemaTables, function createTable(table) {
+                logger.info('Creating table: ' + table);
+                return commands.createTable(table, transaction);
+            }).then(function populateFixtures() {
+                if (tablesOnly) {
+                    return;
+                }
 
-    return tableSequence.then(function () {
-        return fixtures.populate(logger, modelOptions);
+                return fixtures.populate(logger, _.merge({}, {transacting: transaction}, modelOptions));
+            }).then(function commitTransaction() {
+                transaction.commit();
+                resolve();
+            }).catch(function rollbackTransaction(err) {
+                logger.warn('rolling back...');
+                return transaction.rollback(err);
+            });
+        }).catch(function populateDatabaseError(err) {
+            reject(err);
+        });
```

Comment:
This code still looks weird to me, compared to the examples here: http://knexjs.org/#Transactions

Transactions are already promises, so I don't think it needs to be re-wrapped?

Also, I think, if we're going to bother catching the error returned by trx.rollback, then we ought to convert it to a proper error of some sort? Either an InternalServerError or a new DatabaseError of some sort.

E.g. I think the code can be something like:

```
return db.knex.transaction(function populateDatabaseInTransaction(transaction) {
        return Promise.mapSeries(schemaTables, function createTable(table) {
            logger.info('Creating table: ' + table);
            return commands.createTable(table, transaction);
        }).then(function populateFixtures() {
            if (tablesOnly) {
                return;
            }

            return fixtures.populate(logger, _.merge({}, {transacting: transaction}, modelOptions));
        });
    }).catch(function populateDatabaseError(err) {
        return Promise.reject(new errors.InternalServerError('Unable to populate database: ' + err.message));
    });
```